### DISCUSSION TITLE
Using the driver to terminate a protected process programmatically (CLI)

### DISCUSSION THREAD

**Discussion by bladeSk** [winsiderss/systeminformer#1805] [Questions / Answers] [answered]
Is there a way of calling System Informer from the command line or otherwise interfacing with the driver to terminate an overly protected process (MsMpEng.exe)?

It's doable using the GUI, but it's a pain to do this manually every time I need start and stop Defender to make Cisco AnyConnect VPN pass its stupid verification.

**Comment by dmex** [comment, marked as answer]
> Is there a way of calling System Informer from the command line or otherwise interfacing with the driver to terminate an overly protected process (MsMpEng.exe)?

No. System Informer doesn't support actions via command line because they're considered insecure by the industry. The driver also doesn't allow third parties or anything else to 'interface' with the driver for obvious reason.

> It's doable using the GUI

System Informer doesn't support protected process termination, the kernel driver enforces PPL and blocks termination of protected processes. 

> terminate an overly protected process (MsMpEng.exe)?

Process Explorer apparently supports termination of protected processes:
https://github.com/Yaxser/Backstab

Microsoft also includes built-in functionality for termination of protected processes with their SA product that's built-into Windows and included every version since 2000... Network support depends on a SPCR table in the ACPI firmware (generally server hardware) and there's no authentication, anyone with access to the network can remotely terminate protected processes - If you don't have a SPCR table then you only need a serial port on the machine.

**Comment by bladeSk** [comment]
Thanks for pointing me to Backstab, it does exactly what I need.

> System Informer doesn't support protected process termination

Maybe we're talking about a different kind of protected, but I successfully used Process Hacker/System Informer to terminate Windows Defender (using the kernel driver), which is impossible to kill via normal means, even trusted installer can't touch the latest version.

> Microsoft also includes built-in functionality for termination of protected processes with their SA product that's built-into Windows

What does "SA" stand for? I can't find information about this kind of functionality.
